### PR TITLE
Update references of `deploy-sourcegraph-dot-com` to `deploy-sourcegraph-cloud`

### PR DESCRIPTION
**Pull request by danieldides**
The `deploy-sourcegraph-dot-com` repository is being renamed to `deploy-sourcegraph-cloud`. This PR will resolve any references to the old repository.

https://github.com/sourcegraph/sourcegraph/issues/28319

[_Created by Sourcegraph batch change `daniel.dides/rename-dot-com-to-cloud`._](https://k8s.sgdev.org/users/daniel.dides/batch-changes/rename-dot-com-to-cloud)

### CODE REVIEW

**Comment by danieldides** [comment]
I'm not sure we want to update the `internal/cmd/tracking-issue/testdata/fixtures.json` file in this PR. It looks like some sort of historical data.

File: internal/cmd/tracking-issue/testdata/fixtures.json [hunk 1]
```diff
@@ -144,7 +144,7 @@
     {
       "ID": "MDU6SXNzdWU3ODQ3MzY4ODQ=",
       "Title": "code insights: backend MVP implementation",
```
Comment (daxmc99) [inline internal/cmd/tracking-issue/testdata/fixtures.json]: Might want to drop this diff since it could break some test data?

**Review by daxmc99 — CHANGES_REQUESTED** [review]
Looks like a test-data change was picked up here, we should drop that

File: internal/cmd/tracking-issue/testdata/fixtures.json (hunk 1 above)
Comment (danieldides) [reply internal/cmd/tracking-issue/testdata/fixtures.json]: Apologies for all the emails you got from the CLA-bot as I fought with Git on the new computer. This should have been removed from the PR.

**Review by daxmc99 — APPROVED** [review]

**Pull request merged**